xilem_core/src/views/map_state.rs

```diff
@@ -5,12 +5,13 @@ use core::marker::PhantomData;
 
 use crate::{MessageResult, Mut, View, ViewId, ViewMarker, ViewPathTracker};
 
-/// A view that "extracts" state from a [`View<ParentState,_,_>`] to [`View<ChildState,_,_>`].
-/// This allows modularization of views based on their state.
-pub struct MapState<ParentState, ChildState, V, F = fn(&mut ParentState) -> &mut ChildState> {
-    f: F,
+/// The View for [`map_state`] and [`lens`].
+///
+/// See their documentation for more context.
+pub struct MapState<V, F, ParentState, ChildState, Action, Context, Message> {
+    map_state: F,
     child: V,
-    phantom: PhantomData<fn() -> (ParentState, ChildState)>,
+    phantom: PhantomData<fn(ParentState) -> (ChildState, Action, Context, Message)>,
```

Comment:
Good catch, has this kind caused issues already in practice?

Reply:
I don't think so, but I could foresee it being troublesome for the tests.

It might have been fine.

I will add a comment that the variance consequences of this have not been reasoned about.